nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/util/P2DataAccess.java

```diff
@@ -13,124 +13,62 @@
 package org.sonatype.nexus.repository.p2.internal.util;
 
 import java.io.IOException;
-import java.io.InputStream;
-import java.util.List;
+import java.util.Optional;
 
-import javax.annotation.Nullable;
+import javax.inject.Inject;
 import javax.inject.Named;
 
-import org.sonatype.nexus.blobstore.api.Blob;
-import org.sonatype.nexus.common.collect.AttributesMap;
-import org.sonatype.nexus.common.hash.HashAlgorithm;
-import org.sonatype.nexus.repository.Repository;
-import org.sonatype.nexus.repository.storage.Asset;
-import org.sonatype.nexus.repository.storage.AssetBlob;
-import org.sonatype.nexus.repository.storage.Bucket;
-import org.sonatype.nexus.repository.storage.Component;
-import org.sonatype.nexus.repository.storage.MetadataNodeEntityAdapter;
-import org.sonatype.nexus.repository.storage.Query;
-import org.sonatype.nexus.repository.storage.StorageTx;
-import org.sonatype.nexus.repository.view.Content;
-import org.sonatype.nexus.repository.view.Payload;
-import org.sonatype.nexus.repository.view.payloads.BlobPayload;
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.sonatype.goodies.common.Loggers;
+import org.sonatype.nexus.repository.p2.internal.exception.AttributeParsingException;
+import org.sonatype.nexus.repository.p2.internal.metadata.P2Attributes;
+import org.sonatype.nexus.repository.storage.TempBlob;
 
-import com.google.common.base.Supplier;
-import com.google.common.collect.ImmutableList;
-
-import static java.util.Collections.singletonList;
-import static org.sonatype.nexus.common.hash.HashAlgorithm.SHA1;
-import static org.sonatype.nexus.repository.storage.ComponentEntityAdapter.P_VERSION;
-import static org.sonatype.nexus.repository.storage.MetadataNodeEntityAdapter.P_NAME;
+import static com.google.common.base.Preconditions.checkNotNull;
 
 /**
  * Shared code between P2 facets.
  */
 @Named
 public class P2DataAccess
 {
-  public static final List<HashAlgorithm> HASH_ALGORITHMS = ImmutableList.of(SHA1);
+  final Logger log = Loggers.getLogger(this);
```

Comment:
As far as I know you can extend this class from ComponentSupport and use logger from parent